addons/mail/static/src/components/activity/activity_tests.js

```diff
@@ -1150,6 +1150,45 @@ QUnit.test('data-oe-id & data-oe-model link redirection on click', async functio
     );
 });
 
+QUnit.test('after activity edition check the process once activity removed', async function (assert) {
+    assert.expect(3);
+
+    const bus = new Bus();
+    bus.on('do-action', null, payload => {
+        assert.step('do_action');
+        assert.strictEqual(
+            payload.action.context.default_res_id,
+            42,
+            'Action should have the activity res id as default res id in context'
+        );
+        payload.options.on_close();
+    });
+
+    await this.start({
+        env: { bus },
+        mockRPC(route, args) {
+            if (args.method === 'activity_format') {
+                return Promise.resolve([undefined]);
+            }
+            return this._super(...arguments);
+        },
+    });
+    const activity = this.env.models['mail.activity'].create({
+        canWrite: true,
+        id: 12,
+        mailTemplates: [['insert', { id: 1, name: "Dummy mail template" }]],
+        thread: [['insert', { id: 42, model: 'res.partner' }]],
+    });
```

Comment:
I would be best if this used the mock server instead of faking an empty response and creating the activity manually.

This would require to implement in mail mock server:
- `activity_format` (returning proper values)
- `action_feedback` / `action_feedback_schedule_next` (deleting the record)

And then define the test using "this.data" instead of "models.create".

Reply:
Hello @seb-odoo 

Here in 'activity_format', I have returned undefined data because this is only the case when an error is raised.

And 'action_feedback / action_feedback_schedule_next' is not the method which is called during our case, but these methods are calling when we click on 'mark done' from the feedback wizard which is opened on click of 'mark done' from the chatter.

But our case is to click from the form view of mail activity which is opened on click of 'Edit' from the chatter after scheduling an activity.

So, I don't think that it is possible to check the behavior of footer buttons because in the test case we don't have a form view of mail activity.

Can you please give your input on this? Thanks

Reply:
Hello, thank you for looking into it, I have more important tasks that are waiting right now but I'll keep in mind I should check this.